pages/2_DataRAG.py

```
@@ -11,7 +13,59 @@
+            column_config={
+                editable_column: st.column_config.SelectboxColumn(
+                    "STATUS",
+                    options=allowed_values,
+                )
+            },
+            disabled=non_editable_columns,  # Disable editing for other columns
+            key="data_editor",
+        )
+
+        # Button to save changes
+        if st.button("Save Changes"):
```

Ruff (0.8.2)

`st` may be undefined, or defined from star imports

(F405)